test/NuGet.Core.FuncTests/Dotnet.Integration.Test/DotnetListPackageTests.cs

```diff
@@ -302,6 +304,102 @@ public async Task DotnetListPackage_Outdated_Succeed(string currentVersion, stri
+                    $"restore {projectA.ProjectName}.csproj");
+                Assert.True(restoreResult.Success);
+
+                var argsBuilder = new StringBuilder();
+                if (includeTransitive)
+                {
+                    argsBuilder.Append(" --include-transitive");
+                }
+                if (outdated)
+                {
+                    argsBuilder.Append(" --outdated");
+                }
+
+                // Act
+                var listResult = _fixture.RunDotnet(Directory.GetParent(projectA.ProjectPath).FullName,
+                    $"list {projectA.ProjectPath} package {argsBuilder}");
+
+                // Assert
+                if (outdated)
+                {
+                    Assert.Contains("The given project `ProjectA` has no updates given the current sources.", listResult.AllOutput);
+                }
+                else if (includeTransitive)
+                {
+                    Assert.Contains("ProjectB", listResult.AllOutput);
+                }
+                else
+                {
+                    Assert.Contains("No packages were found for this framework.", listResult.AllOutput);
+                }
+            }
+        }
+
+        [PlatformFact(Platform.Windows)]
+        public async Task DotnetListPackage_OutdatedWithNoVersionsFound_Succeeds()
+        {
+            // Arrange
+            using (var pathContext = new SimpleTestPathContext())
+            {
+                var projectA = XPlatTestUtils.CreateProject("ProjectA", pathContext, "net46");
+                var packageX = XPlatTestUtils.CreatePackage(packageId: "packageX", packageVersion: "1.0.0");
+
+                await SimpleTestPackageUtility.CreateFolderFeedV3Async(
+                        pathContext.PackageSource,
+                        PackageSaveMode.Defaultv3,
+                        packageX);
+
+                var addResult = _fixture.RunDotnet(Directory.GetParent(projectA.ProjectPath).FullName,
+                    $"add {projectA.ProjectPath} package packageX --version 1.0.0 --no-restore");
+                Assert.True(addResult.Success);
+
+                var restoreResult = _fixture.RunDotnet(Directory.GetParent(projectA.ProjectPath).FullName,
+                    $"restore {projectA.ProjectName}.csproj");
+                Assert.True(restoreResult.Success);
+
+                foreach (var nupkg in Directory.EnumerateDirectories(pathContext.PackageSource))
+                {
+                    Directory.Delete(nupkg, true);
+                }
+
+                // Act
+                var listResult = _fixture.RunDotnet(Directory.GetParent(projectA.ProjectPath).FullName,
+                    $"list {projectA.ProjectPath} package --outdated");
+
+                // Assert
+                var lines = new List<string>();
+                using (var stringReader = new StringReader(listResult.AllOutput))
```

Comment:
Line-splitting can be done with a lot less code.

    string[] lines = listResult.AllOutput.Split(new [] { Environment.NewLine }, StringSplitOptions.RemoveEmptyEntries);